R/get_publication.R

```diff
@@ -0,0 +1,61 @@
+#' Get publications
+#'
+#' @param page_size Number of results to return in a single query (max 40)
+#' @param page Page number to return
+#' @param verbose Add extra contextual information whilst running
+#'
+#' @return Data frame listing all available publications
+#' @export
+#'
+#' @examples
+#' get_publication_catalogue()
+get_publication_catalogue <- function(
+    page_size = 40,
+    page = NULL,
+    verbose = FALSE) {
+  eesyapi::validate_page_size(page_size)
+  response <- httr::GET(
+    eesyapi::eesapi_url(page_size = page_size, page = page, verbose = verbose)
+  ) |>
+    httr::content("text") |>
+    jsonlite::fromJSON()
+  response |> eesyapi::warning_max_pages()
+  return(response)
+}
+
+#' Get publication specific data set catalogue
+#'
+#' @param publication_id The publication ID as used by the API
+#' @param page_size Number of results to return in a single query (max 40)
```

Comment:
Similar looping / batching of query thing here, I don't think a user cares about pages so we should just batch up multiple queries if we need to and return a full data frame. If a publication has 41 datasets I don't want a user to need to do two different functions to see all and then stack them, or even worse they could miss the last data set altogether.

Would expect the output to be just the $results item but as a data frame

Reply:
Have added in the looping. Kept the page_size and page params for now in case we / users want to use them for any reason, but default behaviour is to get everything.